src/site/content/en/blog/smoothness/index.md

```diff
@@ -0,0 +1,563 @@
+  frameTimes = [...frameTimes.filter(t => t > now - 1000), now];
+  requestAnimationFrame(pollFramesPerSecond);
+  console.log('Frames per second:', frameTimes.length);
+}
+requestAnimationFrame(pollFramesPerSecond);
+```
+
+{% Aside 'warning' %}
+  The code above is an anti pattern and is strongly discouraged!
+{% endAside %}
+
+Using `requestAnimationFrame()` polling is not a good idea for several reasons:
+
+- Every interested party (or script) has to set up its own polling loop.
+- It can block the critical path.
+- Even if the rAF polling function is _fast_, when used continuously it can
+  prevent `requestIdleCallback()` from being able to schedule long idle blocks
+  (blocks that exceed a single frame).
+- Similarly, lack of long idle-blocks prevents the browser from scheduling other
+  long running tasks (such as longer garbage collection and other background or speculative
+  work).
+- If polling is toggled on and off, then you'll miss cases where frame budget has
+  been exceeded.
+- Polling will report false-positives in cases where the browser is using
+  variable update frequency (for example, due to power or visibility status).
+- And most importantly, it doesn't actually capture all types of animation
+  updates!
+
+You can see how too much work on the main thread can impact the ability to see
+animation frames. You can play with the [Jank
+Sample](https://googlechrome.github.io/devtools-samples/jank/) to see how a rAF
+driven animation, once there is too much work on the main thread work (i.e.
+layout) will lead to dropped frames and fewer rAF callbacks, and lower fps.
+
+When the main thread becomes bogged down, visual updates begin to stutter.
+**Jank!**
+
+Many measurement tools have focused extensively on the ability for the main
+thread to yield in a timely manner, and for animation frames to run smoothly.
+But this is not the whole story! Consider the following example:
+
+**TODO(mmocny): add video link.**
+
+This video shows a page that periodically injects long tasks onto the main
+thread. These long tasks completely ruin the ability for the page to provide
+certain types of visual updates, and you can see a corresponding drop of
+`requestAnimationFrame()` reported FPS to 0.
+
+And yet, the page continues to scroll smoothly. This is because on modern
+browsers, [scrolling is often
+threaded](https://blogs.windows.com/msedgedev/2017/03/08/scrolling-on-the-web/#:~:text=The%20multithreaded%20web),
+driven entirely by the compositor.
+
+This is an example that simultaneously contains many dropped frames on the main
+thread, yet still has many successfully-delivered frames of scrolling on the
+compositor thread. What's more, the main thread paint update, once the long task
+is complete, actually has no visual change to offer anyway. rAF polling
+suggested a frame drop to 0, but _visually_, a user wouldn't be able to notice a
+difference!
+
+As you can see, for animation frames, the story is just not that simple.
```

Comment:
Hey now, I had used the phrase "simply not that simple" and I was proud of that joke.